buildlib/azure-pipelines-perf.yml

```diff
@@ -128,5 +130,11 @@ stages:
             clean: true
           - bash: |
               set -x
+              echo 'Retain 3 latest workdirs for manual repro; clean the rest'
+              wrk_dirs="/hpc/scrap/azure/$(Build.DefinitionName)/"
+              for dir in $(ls -lt "$wrk_dirs" | tail -n +4 | awk '{print $NF}'); do
+                  echo "Removing old workdir: $wrk_dirs$dir"
+                  rm -rf "$wrk_dirs$dir"
```

Comment:
Consider changing to more readable version:
`rm -rf "${wrk_dirs}/${dir}"`

Reply:
The syntax suggested raises shellcheck [SC2115](https://www.shellcheck.net/wiki/SC2115)

Reply:
Some extra curly brackets never hurt though, added.